src/u_websocket.c

```diff
@@ -26,6 +26,16 @@
 #include "u_private.h"
 #include "ulfius.h"
 
+// MSG_NOSIGNAL is not defined on OSX
+#ifdef __APPLE__
+  #include "TargetConditionals.h"
+  #ifdef TARGET_OS_MAC
```

Comment:
Isn't it redundant having both `#ifdef __APPLE__` and `#ifdef TARGET_OS_MAC` in this case?

Also, is `"TargetConditionals.h"` always present in apple devices?

Reply:
`_APPLE__` is defined for iPhone also that's why I specified `TARGET_OS_MAC`.

"TargetConditionals.h" is supposed to be present on all OSX platforms, that what I understand from 

```
TargetConditionals.h in 3.4 Universal Interfaces works
                        with all compilers.  This header only recognizes compilers
                        known to run on Mac OS X.
```

from : https://opensource.apple.com/source/CarbonHeaders/CarbonHeaders-18.1/TargetConditionals.h 

My solution is heavily based on this post : https://stackoverflow.com/questions/5919996/how-to-detect-reliably-mac-os-x-ios-linux-windows-in-c-preprocessor/5920028#5920028

Reply:
Also according to https://stackoverflow.com/questions/5919996/how-to-detect-reliably-mac-os-x-ios-linux-windows-in-c-preprocessor/5920028#5920028 I changed `#ifdef TARGET_OS_MAC` to `#if TARGET_OS_MAC`

Reply:
For example, if I take inspiration from this PR: https://github.com/libretro/dolphin/pull/41/files , the change would simply be:

```C
#if define(__APPLE__) && !define (MSG_NOSIGNAL)
  #define MSG_NOSIGNAL 0
#endif
```

What about it?

Reply:
Done !

Tested, it builds ok on OSX

Reply:
cool, thanks!